src/main/kotlin/bakalarka/elearningplatform/service/ChapterService.kt

```diff
@@ -0,0 +1,72 @@
+package bakalarka.elearningplatform.service
+
+import bakalarka.elearningplatform.db.CourseChapterRepository
+import bakalarka.elearningplatform.db.CourseRepository
+import bakalarka.elearningplatform.model.Course
+import bakalarka.elearningplatform.model.CourseChapter
+import bakalarka.elearningplatform.request.AddCourseChapterRequest
+import bakalarka.elearningplatform.security.Management
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+
+@Service
+class ChapterService(
+    var courseChapterRepository: CourseChapterRepository,
+    var courseRepository: CourseRepository,
+    var userService: UserService,
+    var management: Management
+) {
+
+    fun addChapter(request: AddCourseChapterRequest, courseId: Long): CourseChapter {
+        val (chapterTitle, description, content, position) = request
+        val course = courseRepository.findById(courseId)
+        val chapter = courseChapterRepository.save(
+            CourseChapter(
+                chapterTitle = chapterTitle,
+                description = description,
+                content = content,
+                position = position,
+                course = courseRepository.findByIdOrNull(courseId)
+            )
+        )
+        if (course.isPresent) {
+            course.get().courseChapters.add(position, chapter)
```

Comment:
Tomuto nerozumiem - to by som ocakaval, ze sa nejak deje samo - tu by nemali byt podla mna potrebne nejake dalsie operacie. Alebo aspon pri pridavani by malo vzdy pridat na koniec, ci? Hm.

Reply:
Tam ide o situaciu, kedy sa vytvarany chapter ma zaradit niekde inde ako nakoniec.. zrejme by sa tam patrila podmienka nech sa to nevykonava pokial je chapter normalne zaradeny na konci

Reply:
Nepaci sa mi na tom nejak, ze najprv to tam pridas, potom removnes last (preco last) a potom updatujes ordery. Nie je to lahko pochopitelne. Momentalne nemam riesenie, skus sa ked tak zamysliet, ci sa to neda nejak zjednodusit a ja skusim neskor (zajtra) tiez. Nie je to big deal, ale bolo by fajn to spravit nejak krajsie :wink: